chall09/bazuara.c

```diff
@@ -0,0 +1,131 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   bazuara.c                                          :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: bazuara <bazuara@student.42madrid.>        +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2020/05/09 14:07:16 by bazuara           #+#    #+#             */
+/*   Updated: 2020/05/09 20:13:40 by bazuara          ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+
+char	*ft_strjoin(char const *s1, char const *s2)
+{
+	char	*new;
+	size_t	total;
+	size_t	i;
+	size_t	j;
+
+	i = 0;
+	if (!s1 || !s2)
+		return (0);
+	total = strlen(s1) + strlen(s2);
+	if ((new = malloc(((total) * sizeof(char)) + 1)))
+	{
+		while (i < strlen(s1))
+		{
+			new[i] = s1[i];
+			i++;
+		}
+		j = 0;
+		while (j < strlen(s2))
+		{
+			new[i + j] = s2[j];
+			j++;
+		}
+		new[i + j] = '\0';
+	}
+	return (new);
+}
+
+char		*ft_strjoin_free(char *str1, char *str2, int i)
+{
+	char *temp;
+
+	temp = ft_strjoin(str1, str2);
+	if (i == 1)
+		free(str1);
+	else if (i == 2)
+		free(str2);
+	if (i == 3)
+	{
+		free(str1);
+		free(str2);
+	}
```

Comment:
A little advice on making this shorter and easier to read
```suggestion
	if (i == 1 || i == 3)
		free(str1);
	if (i == 2 || i == 3)
		free(str2);
```